### PR TITLE
[Workload controller manager needs to die sooner #174]

### PR DESCRIPTION
**Pull request by Hong-Chang**
What type of PR is this?
/kind feature

What this PR does / why we need it:
Currently, controller instance record in registry will be automatically deleted after 5 minutes when it could not renew its lease. This is designed for a very stable environment. However, initial testing often has stability problem and caused workload controller manager restart. Further investigation also found api server does lease renewal every 10 seconds. This change will allow problematic controller instances to be discovered much faster.

Which issue(s) this PR fixes:
Workload controller manager needs to die sooner #174

Special notes for your reviewer:
Please suggest configured values.
I introduced type ControllerConfig, so renamed original ControllerConfig to ControllerConfigMap.

Does this PR introduce a user-facing change?:
No

### CODE REVIEW

File: cmd/workload-controller-manager/app/config/controllerconfig.go [hunk 1]
```diff
@@ -23,7 +23,8 @@ import (
 )
 
 type controllerTypes struct {
```
Comment (Sindica) [inline cmd/workload-controller-manager/app/config/controllerconfig.go:25]: Naming is not fit.

**Comment by Sindica** [comment]
Joe initially authored controller config. Adding him for review.

File: cmd/workload-controller-manager/config/controllerconfig.json [hunk 2]
```diff
@@ -1,4 +1,5 @@
 {
+    "reportHealthIntervalInSecond": 10,
```
Comment (jshaofuturewei) [inline cmd/workload-controller-manager/config/controllerconfig.json:2]: The config here looks like the controllers of all the types share the same reportHealthIntervalInSecond but in the codes each controller type deals with reportHealthIntervalInSecond  seperately.

File: cmd/workload-controller-manager/app/config/controllerconfig.go (hunk 1 above)
Comment (Hong-Chang) [reply cmd/workload-controller-manager/app/config/controllerconfig.go]: Renamed type to controllerManagerConfig

File: cmd/workload-controller-manager/config/controllerconfig.json (hunk 2 above)
Comment (Hong-Chang) [reply cmd/workload-controller-manager/config/controllerconfig.json:2]: Yes, reportHealthIntervalInSecond has one value for all types. Its usage is when starting each controller so appears to be seperately. I think it's ok.

**Pull request closed without merging**